src/DocumentPicker.Sample/DocumentPickerPage.xaml.cs

```diff
@@ -2,60 +2,74 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using System.IO;
 using DocumentPicker.Plugin.Shared;
-using Xamarin.Essentials;
+using FFImageLoading;
+using Svg;
+using Svg.Platform;
 using Xamarin.Forms;
 using Xamarin.Forms.Xaml;
+using FileSystem = Xamarin.Essentials.FileSystem;
+using Image = Xamarin.Forms.Image;
+using System.Data.SqlTypes;
+using Color = Svg.Interfaces.Color;
 
 namespace DocumentPicker.Samples
 {
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class DocumentPickerPage : ContentPage
     {
-        ObservableCollection<FileResult> files = new ObservableCollection<FileResult>();
+
+        private readonly DocumentConverterService _converterService = new DocumentConverterService();
+
+        ObservableCollection<DocumentResult> files = new ObservableCollection<DocumentResult>();
         public DocumentPickerPage()
         {
             InitializeComponent();
 
-            //files.CollectionChanged += Files_CollectionChanged;
+            files.CollectionChanged += Files_CollectionChanged;
 
             pickDocument.Clicked += async (sender, args) =>
             {
-                var x = new DocumentPickerService();
+                var fileResult = await _converterService.PickDocumentAsync();
 
-                var fileResult = await x.PickDocumentAsync();
-                
                 files.Add(fileResult);
+
             };
         }
 
-   //     private void Files_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
-   //     {
-   //         if (files.Count == 0)
-   //         {
-   //             ImageList.Children.Clear();
-   //             return;
-   //         }
-   //         if (e.NewItems.Count == 0)
-   //             return;
-
-   //         var file = e.NewItems[0] as DocumentFile;
-   //         var image = new Image { WidthRequest = 300, HeightRequest = 300, Aspect = Aspect.AspectFit };
-   //         if (file != null)
-   //         {
-   //             image.Source = ImageSource.FromFile(file.FileResult.FullPath);
-   //             /*image.Source = ImageSource.FromStream(() =>
-   //{
-   //    var stream = file.GetStream();
-   //    return stream;
-   //});*/
-   //             ImageList.Children.Add(image);
-
-   //             var image2 = new CachedImage { WidthRequest = 300, HeightRequest = 300, Aspect = Aspect.AspectFit };
-   //             image2.Source = ImageSource.FromFile(file.FileResult.FullPath);
-   //             ImageList.Children.Add(image2);
-   //         }
-   //     }
+        private void Files_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
```

Comment:
converting to SVG may take quite some time.
So try this in your sample (as you will need it in icl.filler as well)
- make this method `async`
- wrap the lines 53 to 62 in `await Task.Run(()=>)` as this runs stuff in the background
- `task.run()` should return the path to the image.
- this path is then usedin line 64

As a bonus - show a progress indicator (some progress circle) as long as the conversion runs, so the user realizes, there is something going on!